ui/components/multichain/pages/permissions-page/connection-list-item.js

```diff
@@ -27,14 +33,22 @@ import { getURLHost } from '../../../../helpers/utils/util';
 import SnapAvatar from '../../../app/snaps/snap-avatar/snap-avatar';
 import { AvatarGroup } from '../../avatar-group';
 import { AvatarType } from '../../avatar-group/avatar-group.types';
+import { getUseBlockie } from '../../../../selectors';
 
 export const ConnectionListItem = ({ connection, onClick }) => {
   const t = useI18nContext();
   const isSnap = connection.subjectType === SubjectType.Snap;
   const AVATAR_GROUP_LIMIT = 5;
-  const addressIconList = connection.addresses?.map((address) => ({
-    avatarValue: address,
-  }));
+  const TOOLTIP_LIMIT = 7;
```

Comment:
This logic should move to a new component ConnectedAccountsTooltip component to handle the limit